apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -38,7 +38,7 @@ namespace Google.Cloud.Spanner.V1
     /// and it will incur a penalty on other Spanner processes because it will be an hour before the server frees the session if 
     /// it's not properly deleted.
     /// </summary>
-    internal static class SessionPool
+    public static class SessionPool
```

Comment:
Maybe instead of this:
        private readonly ApiCall<CreateSessionRequest, Session> _callCreateSession;
...
            _callCreateSession = clientHelper.BuildApiCall<CreateSessionRequest, Session>(
                GrpcClient.CreateSessionAsync, GrpcClient.CreateSession, effectiveSettings.CreateSessionSettings);

we could have this:
        private readonly IApiCall<CreateSessionRequest, Session> _callCreateSession; <=IApiCall
...
            _callCreateSession = new PassThroughProxy( clientHelper.BuildApiCall<CreateSessionRequest, Session>(
                GrpcClient.CreateSessionAsync, GrpcClient.CreateSession, effectiveSettings.CreateSessionSettings));
             Modify_CallProxy(nameof(GrpcClient.CreateSessionAsync), ref _callCreateSession);

Then, I could implement the partial method Modify_CallProxy to wrap the underlying api call to do either exception filtering, pooling, or whatever.